src/crypto/polynomials.rs

```diff
@@ -0,0 +1,319 @@
+use frost_core::{
+    keys::CoefficientCommitment, serialization::SerializableScalar, Field, Group, Scalar,
+};
+use rand_core::CryptoRngCore;
+
+use super::ciphersuite::Ciphersuite;
+use crate::{
+    participants::ParticipantMap,
+    protocol::{Participant, ProtocolError},
+};
+
+use std::ops::Add;
+
+pub struct Polynomial<C: Ciphersuite>(Vec<Scalar<C>>);
+
+impl<C: Ciphersuite> Polynomial<C> {
+    /// Constructs the polynomial out of scalars
+    /// The first scalar (coefficients[0]) is the constant term
+    pub fn new(coefficients: Vec<Scalar<C>>) -> Self {
+        Polynomial(coefficients)
+    }
+
+    pub fn get_coefficients(&self) -> Vec<Scalar<C>> {
+        self.0.to_vec()
+    }
+
+    /// Outputs the degree of the polynomial
+    pub fn degree(&self) -> usize {
+        let mut degree = self.0.len();
+        // loop as long as the higher terms are zero
+        while degree > 0 && self.0[degree - 1] == <C::Group as Group>::Field::zero() {
+            degree -= 1;
+        }
+        if degree == 0 {
+            0
+        } else {
+            degree - 1
+        }
+    }
+
+    /// Creates a polynomial p of degree threshold - 1
```

Comment:
The polynomial created is of degree "degree", this method seems to be independent from the threshold
```suggestion
    /// Creates a random polynomial p of the given degree
```